tests/integration/article.test.js

```diff
@@ -224,60 +225,330 @@ describe("# Articles endpoints", () => {
+              .to.haveOwnProperty("username")
+              .to.be.a("string");
+          });
+          done();
+        });
+    });
+
+    describe("Pagination (limit the number of articles to be returned)", () => {
+      it("should return the exact number of articles specified by the limit param", done => {
+        chai
+          .request(app)
+          .get(`/api/v1/articles?limit=2`)
+          .end((err, res) => {
+            expect(res.status).equals(OK);
+            expect(res.body.message).to.contain("Successful");
+            expect(res.body)
+              .to.haveOwnProperty("articles")
+              .to.be.an("array");
+            expect(res.body)
+              .to.haveOwnProperty("articlesCount")
+              .to.be.a("number");
+            expect(res.body.articles.length).to.equal(2);
+            expect(res.body.articlesCount).to.equal(2);
+            done();
+          });
       });
-  });
-  it("should be ready to be posted on linkedIn", done => {
-    chai
-      .request(app)
-      .get(`/api/v1/articles/${validSlug}/share/linkedIn`)
-      .set("Authorization", `Bearer ${token}`)
-      .end((err, res) => {
-        expect(res.status).equals(OK);
-        expect(res.body.message).to.contain(
-          "Article ready to be posted on linkedIn"
-        );
-        done();
+
+      it("should return (x) articles for the first page, and y articles for the second page", done => {
+        chai
+          .request(app)
+          .get(`/api/v1/articles?limit=1&page=1`)
+          .end((err, page1) => {
+            chai
+              .request(app)
+              .get(`/api/v1/articles?limit=2&page=2`)
+              .end((error, page2) => {
+                expect(page1.status).equals(OK);
+                expect(page1.body.message).to.contain("Successful");
+                expect(page1.body)
+                  .to.haveOwnProperty("articles")
+                  .to.be.an("array");
+                expect(page1.body)
+                  .to.haveOwnProperty("articlesCount")
+                  .to.be.a("number");
+                expect(page1.body.articles.length).to.equal(1);
+                expect(page1.body.articlesCount).to.equal(1);
+                expect(page2.status).equals(OK);
+                expect(page2.body.message).to.contain("Successful");
+                expect(page2.body)
+                  .to.haveOwnProperty("articles")
+                  .to.be.an("array");
+                expect(page2.body)
+                  .to.haveOwnProperty("articlesCount")
+                  .to.be.a("number");
+                expect(page2.body.articles.length).to.equal(2);
+                expect(page2.body.articlesCount).to.equal(2);
+                expect(page1.body.articles).not.to.equal(page2.body.articles);
+                done();
+              });
+          });
       });
+    });
+  });
+
+  describe("Updating and deleting a specific article", () => {
+    before(done => {
```

Comment:
can you try to use the token and the user we have created in setups.test file instead of creating a new user ?
It can help us to increase performance.

Reply:
@espoirMur, if I use that user to test the delete API, it will mark the user as deleted and the next test case scenarios that are using the same user can be affected. That's the reason behind creating another user to test the delete API.